examples/codeviewer/common/src/jvmMain/kotlin/org/jetbrains/codeviewer/platform/JvmFile.kt

```diff
@@ -58,22 +64,36 @@ fun java.io.File.toProjectFile(): File = object : File {
             override val size get() = size
 
             override fun get(index: Int): String {
-                val startPosition = lineStartPositions[index]
-                val length = if (index + 1 < size) lineStartPositions[index + 1] - startPosition else
-                    byteBufferSize - startPosition
-                // Only JDK since 13 has slice() method we need, so do ugly for now.
-                byteBuffer.position(startPosition)
-                val slice = byteBuffer.slice()
-                slice.limit(length)
+                val position = lineRange(index)
+                val slice = byteBuffer.slice(position.first, position.last - position.first)
                 return StandardCharsets.UTF_8.decode(slice).toString()
             }
+
+            private fun lineRange(index: Int): IntRange {
+                val startPosition = lineStartPositions[index]
+                val nextLineIndex = index + 1
+                var endPosition = if (nextLineIndex < size) {
+                    // Remove \n itself from the range
+                    lineStartPositions[nextLineIndex] - 1
+                } else byteBufferSize
+
+                // Clip windows style line ending from the range too
+                if (byteBuffer[endPosition - 1].isChar('\r')) {
```

Comment:
```suggestion
                if (endPosition > 0 && byteBuffer[endPosition - 1].isChar('\r')) {
```

endPosition can be zero after the previous substraction (for a file with unix-like line endings)

Reply:
No, it can't.
Look at `lineStartPositions[nextLineIndex] - 1`:
- `nextLineIndex` always > 0
- Zero value can be only first element of `lineStartPositions`

Reply:
Yep. But I guess `endPosition > startPosition` would work even better